source/docs/software/commandbased/commands-v3/structuring-v3-project.rst

```diff
@@ -0,0 +1,465 @@
+       public static final int RIGHT_LEADER_ID = 2;
+       public static final double MAX_SPEED = 4.0;
+     }
+
+     public static final class Arm {
+       public static final int MOTOR_ID = 5;
+       public static final double HIGH = 1.57;
+       public static final double MID = 0.78;
+       public static final double LOW = 0.0;
+     }
+   }
+
+## Priority Scheme
+
+Document your priority levels:
+
+.. code-block:: java
+
+   public final class Priorities {
+     public static final int DEFAULT = 0;           // Normal commands
+     public static final int AUTO_ACTION = 10;      // Automated actions
+     public static final int DRIVER_OVERRIDE = 50;  // Manual overrides
+     public static final int SAFETY = 500;          // Safety interlocks
+     public static final int EMERGENCY = 1000;      // Emergency stops
+   }
+
+Use these constants:
+
+.. code-block:: java
+
+   Command eStop = drivetrain.run(coro -> drivetrain.stop())
+     .withPriority(Priorities.EMERGENCY)
+     .named("Emergency Stop");
+
+## Telemetry
+
+Put telemetry in ``periodic()`` methods:
+
+.. code-block:: java
+
+   @Override
+   public void periodic() {
+     SmartDashboard.putNumber("Drivetrain/Left Distance", getLeftDistance());
+     SmartDashboard.putNumber("Drivetrain/Right Distance", getRightDistance());
+     SmartDashboard.putData("Drivetrain/Odometry", odometry);
+
+     // Use mechanism name as prefix for organization
+   }
+
+## Testing
+
+Create test commands for individual mechanisms:
+
+.. code-block:: java
+
+   public class TestCommands {
+
+     public static Command testDrivetrain(Drivetrain drivetrain) {
+       return drivetrain.run(coroutine -> {
+         System.out.println("Testing drivetrain...");
+         drivetrain.tank(0.3, 0.3);
+         coroutine.wait(Seconds.of(1.0));
+         drivetrain.stop();
+         System.out.println("Drivetrain test complete");
+       }).named("Test Drivetrain");
+     }
+
+     public static Command testArm(Arm arm) {
+       return arm.run(coroutine -> {
+         System.out.println("Testing arm...");
+         coroutine.await(arm.moveTo(Constants.ARM_MID));
+         coroutine.wait(Seconds.of(0.5));
+         coroutine.await(arm.moveTo(Constants.ARM_LOW));
+         System.out.println("Arm test complete");
+       }).named("Test Arm");
+     }
+   }
+
+Bind test commands to buttons:
+
+.. code-block:: java
+
+   // In RobotContainer.configureBindings()
+   driver.povUp().onTrue(TestCommands.testDrivetrain(drivetrain));
+   driver.povDown().onTrue(TestCommands.testArm(arm));
+
+## Common Mistakes to Avoid
+
+1. **Scheduling commands inside mechanisms**: Return commands from methods, don't schedule them
+
+.. code-block:: java
+
+   // ❌ BAD: Scheduling inside mechanism
+   public void grab() {
+     run(coro -> { /* ... */ }).schedule();
+   }
+
+   // ✅ GOOD: Return a command
+   public Command grab() {
+     return run(coro -> { /* ... */ }).named("Grab");
```

Comment:
We should be consistent with coroutine parameter names. i generally prefer `coroutine` for explicit examples or just `co` for conciseness